src/fsharp/nameres.fs

```diff
@@ -2559,24 +2575,31 @@ let ResolveCompletionsInType (ncenv: NameResolver) nenv isApplicableMeth m ad st
         result
 
     let pinfoItems = 
+        let pinfos = 
+            match completionTargets with
+            | ResolveCompletionTargets.SettablePropertiesAndFields -> pinfos |> List.filter (fun p -> p.HasSetter)
+            | _ -> pinfos
+
         pinfos
         |> List.map (fun pinfo -> DecodeFSharpEvent [pinfo] ad g ncenv m)
-        |> List.filter (fun pinfo-> match pinfo with
-                                    | Some(Item.Event(einfo)) -> IsStandardEventInfo ncenv.InfoReader m ad einfo
-                                    | _ -> pinfo.IsSome)
+        |> List.filter (fun pinfo->
+            match pinfo with
+            | Some(Item.Event(einfo)) -> completionTargets.ResolveAll && IsStandardEventInfo ncenv.InfoReader m ad einfo
+            | _ -> pinfo.IsSome)
         |> List.map (fun pinfo->pinfo.Value)
```

Comment:
instead of 
```fsharp
  |> List.filter (fun pinfo->
    match pinfo with
    | Some(Item.Event(einfo)) -> completionTargets.ResolveAll && IsStandardEventInfo ncenv.InfoReader m ad einfo
    | _ -> pinfo.IsSome)
  |> List.map (fun pinfo-> pinfo.Value)
```

you can use List.choose to remove `.IsSome`, `.Value` antipattern
```fsharp
  |> List.choose (fun pinfo->
    match pinfo with
    | Item.Event(einfo) -> if completionTargets.ResolveAll && IsStandardEventInfo ncenv.InfoReader m ad einfo then Some pinfo else None
    | _ -> Some pInfo
```